src/Resource/CloudAccount/Endpoint.php

```diff
@@ -148,6 +151,139 @@ protected function _waitUntilPhpVersion(
     };
   }
 
+  /**
+   * Create a backup
+   *
+   * @param Entity An instance of cloud account entity.
+   * @return Backup
+   * @throws ApiException If request fails
+   */
+  public function createBackup(Entity $entity) : Backup {
+    $this->wait(null);
+    $response = $this->_client->request(
+      'POST',
+      self::_URI . "/{$entity->getId()}/backup"
+    );
+
+    return $this->getModel(Backup::class)->sync($response);
+  }
+
+  /**
+   * Return a list of backups
+   *
+   * @return Backup
+   * @throws ApiException If request fails
+   */
+  public function getBackups(Entity $entity) : Collection {
+    $this->wait(null);
+    $collection = new Collection(Backup::class);
+    
+    foreach ($this->_fetchBackupList($entity) as $backup) {
+      $collection->add($this->getModel(Backup::class)->sync($backup));
+    }
+
+    return $collection;
+  }
+
+  /**
+   * Return a specific backup
+   *
+   * @param string $file_name The unique file name for the backup to retrieve.
+   * @return Backup
+   * @throws ApiException If request fails
+   */
+  public function getBackup(Entity $entity, string $file_name) : Backup {
+    $this->wait(null);
+    return $this->_findBackup($entity, $file_name);
+  }
+
+  /**
+   * Download a specific backup
+   *
+   * @param string $file_name The unique file name for the backup to retrieve.
+   * @param string $path the directory to store the download in.
+   *
+   * @return Promise
+   * @throws ApiException If request fails
+   * @throws Exception
+   */
+  public function downloadBackup(Entity $entity, string $file_name, string $path) {
+    $this->wait(null);
+
+    if (! file_exists($path) || ! is_dir($path)) {
+      throw new Exception('##LG_INVALID_PATH##');
+    }
+
+    $path = trim($path);
+    if (substr($path, -1) !== DIRECTORY_SEPARATOR) {
```

Comment:
in other places i do `$path = rtrim($path, '/')` and then explicitly add the DS like `"$path/$name"`

also, do you think it's worth using `DIRECTORY_SEPARATOR` over just `/`? afaik php uses `/` correctly in windows.

Reply:
I wasn't aware that internally PHP had started translating directory separators. Since it's still considered a "best practice" (or common practice) to sue DIRECTORY_SEPARATOR, I'd rather use that. Esp since this is an external project.

The IF statements makes sure that we handle 'some/path ' as well as 'some/path/' Basically I'm trying to say trim the string and then if the last chr is not a DS, add a DS.